packages/manager-react-components/src/components/filters/filter-add.component.tsx

```diff
@@ -82,31 +96,23 @@ export function FilterAdd({ columns, onAddFilter }: Readonly<FilterAddProps>) {
               {t('common_criteria_adder_operator_label')}
             </span>
           </div>
-          {selectedColumn &&
-            columns.map((column) => {
-              return (
-                <div
-                  key={`filter-condition-select-${column.id}`}
-                  className={column.id === selectedColumn.id ? '' : 'hidden'}
-                >
-                  <OdsSelect
-                    name={`add-operator-${column?.id}`}
-                    value={selectedComparator}
-                    onOdsChange={(event) => {
-                      setSelectedComparator(
-                        event.detail.value as FilterComparator,
-                      );
-                    }}
-                  >
-                    {column?.comparators?.map((comp) => (
-                      <option key={`${column.id}-${comp}`} value={comp}>
-                        {t(`common_criteria_adder_operator_${comp}`)}
-                      </option>
-                    ))}
-                  </OdsSelect>
-                </div>
-              );
-            })}
+          {selectedColumn && (
+            <div key={`filter-condition-select-${selectedColumn.id}`}>
+              <OdsSelect
+                name={`add-operator-${selectedColumn?.id}`}
+                value={selectedComparator}
+                onOdsChange={(event) => {
+                  setSelectedComparator(event.detail.value as FilterComparator);
+                }}
+              >
+                {selectedColumn?.comparators?.map((comp) => (
+                  <option key={`${selectedColumn.id}-${comp}`} value={comp}>
```

Comment:
```suggestion
                  <option key={comp} value={comp}>
```
Key is not an id, you only need the key of your element inside the list